### PR TITLE
update startup scripts

### PR DESCRIPTION
**Pull request by danielpeach**
@ewiseblatt or @anotherchrisberry 

This should clear up most of the existing issues with running Deck either locally or on our vms.

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+    echo $npm_package_engines_node > $PWD/.nvmrc
+  elif [[ $(cat $PWD/.nvmrc) != $npm_package_engines_node ]]; then
+    echo "WARNING: Your .nvmrc specifies node $(cat $PWD/.nvmrc)."
+  fi
+}
+
 if [[ $NODE_VERSION != $npm_package_engines_node ]]; then
-  if [[ -f $HOME/.nvm/nvm.sh ]]; then
-    echo "Updating your node version to $npm_package_engines_node..."
+  if [[ -f /etc/profile.d/nvm.sh ]]; then
```
Comment (ewiseblatt) [inline start.sh]: maybe, but should the precedence be reversed so a user trumps an installation?

File: start.sh [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
+if [[ -z "$npm_package_engines_node" ]]; then
+  echo "You are probably running this script without NPM. Try running 'npm start' from the Deck directory root."
+  exit
+fi
+
 NODE_VERSION=$(node -v)
 
+function update_node() {
+  echo "Updating your node version to $npm_package_engines_node..."
+  nvm use $npm_package_engines_node
```
Comment (ewiseblatt) [inline start.sh]: I prefer if ! nvm use ..

File: start.sh [hunk 3]
```diff
@@ -1,19 +1,37 @@
 #!/bin/bash
 
+if [[ -z "$npm_package_engines_node" ]]; then
+  echo "You are probably running this script without NPM. Try running 'npm start' from the Deck directory root."
+  exit
```
Comment (ewiseblatt) [inline start.sh]: exit with failure

**Comment by icfantv** [comment]
Just my one question.  I don't know that it's always a good idea to show a warning message to a user if the version they're using is _higher_ than the minimum.

**Comment by icfantv** [comment]
@danielpeach hey, what's the status on this?

**Comment by danielpeach** [comment]
I'm going to close this for now, since it doesn't fully solve what I've been intending to do, which is require that `npm install` runs with the proper version of node.

**Pull request closed without merging**